Cargo.toml

```diff
@@ -46,6 +46,10 @@ rand_core = "0.4"
 serde_test = "1.0"
 bitcoin_hashes = "0.7"
 
+[dependencies.bitcoin_hashes]
+version = "0.7"
+optional = true
```

Comment:
Any arguments here against making that default? Annoyingly someone that is depending on `bitcoin` and depends on secp and hashes using `bitcoin::secp256k1` and `bitcoin::hashes` can't add features to the secp dependency. Well, that's a broader Cargo problem, but well.

Reply:
I thought about this _a lot_ :sweat_smile: (also because of features I might want to activate in `secp256k1/bitcoin_hashes` to not cause incompatibilities), turns out that cargo builds dependencies with the union of all the features activated if a dependency exists multiple times in the dependency graph (just tested it with a minimal example). So all they had to do was to specify `secp256k1` as a dependency themselves and activate the feature.

On the other hand we could just activate that feature in `bitcoin` ~and use it there too~. After taking a closer look we don't seem to have any transaction signing code? At least we don't use `secp256k1::Message` _anywhere_.

Reply:
Sure? It might be that we use an `into()` and that `Message` is never mentioned verbatim. I add a usage of this in the [`Bitcoin Signed Message PR`](https://github.com/rust-bitcoin/rust-bitcoin/pull/413).